build/documentation/copy-package-readme.js

```diff
@@ -29,36 +31,130 @@ process.argv.forEach(function (val, index) {
+        srcFiles.forEach((srcReadmePath) => {    
+
+            const destReadmePath = srcReadmePath.replace(/(\bpackages\b)(?!.*\1)/, destDir);
             const dirPath = path.dirname(destReadmePath);
+            const srcDirPath = path.dirname(srcReadmePath);
+        
+            var title = srcDirPath
+                .split(path.sep)
+                .pop()
+        
+            var msg = '';
+
             if (!fs.existsSync(dirPath)) {
-                dryRun ? console.log("----> Would create folder " + dirPath) : fs.mkdirSync(dirPath);
+                msg += "      ...CREATE folder '" + title + "'\n";
             }
-            if (!fs.existsSync(destReadmePath)) {
-                dryRun ? console.log("  --> Would create file README.md in " + dirPath) : fs.copyFileSync(filePath, destReadmePath);
-            } else {
-                dryRun ? console.log("  --> Would replace README.md in " + dirPath) : fs.copyFileSync(filePath, destReadmePath);
+            if (!fs.existsSync(destReadmePath)) {
+                msg += "       ...ADD readme.md into the '" + title + "' folder";
+            } else {
+                msg += "       ...REPLACE readme.md in the '" + title + "' folder";
             }
-            
+
+            console.log(msg);
+
         });
 
+        console.log("   In website/sidebars.json, this script updates...\n" +
+        "      ...the Packages array with the filepath to each package's index file.");
+    });
+
+}
+
+/**
+ * Function to copy readme files to the ./docs/en/packages folder
+ * and update Docusaurus sidebar, ./website/sidebars.json
+ */
+function copyReadmeFiles() {
+
+    const resolvedSrcReadmePaths = path.resolve(rootDir, srcReadmePaths);
+    glob(resolvedSrcReadmePaths, {realpath:true}, function(error, srcFiles) {
+        srcFiles.forEach((srcReadmePath) => {    
+            createDestReadme(srcReadmePath);
+            const srcDirPath = path.dirname(srcReadmePath);
+            sidebarEntries.push("en/packages/" + srcDirPath.split(path.sep).pop() + "/index");
+        });
+        updateSidebar();
     });
 
 }
 
 /**
- * Copy all files
+ * Builds a new readme in ./docs/en/packages
+ * Creates and adds a docusaurus header to the new file
+ * Then appends the original readme from .docs/en/packages/[package-name]
+ */
+function createDestReadme(srcReadmePath) {
+ 
+    const destReadmePath = srcReadmePath.replace(/(\bpackages\b)(?!.*\1)/, destDir);
+    const destDirPath = path.dirname(destReadmePath);
+    const srcDirPath = path.dirname(srcReadmePath);
+    const srcReadmeText = fs.readFileSync(srcReadmePath).toString();
+
+    var title = srcDirPath
+                    .split(path.sep)
+                    .pop()
+                    .replace(/-/g, ' ')
+                    .replace(/fast /g, '')
+                    .toLowerCase()
+                    .split(' ')
+                    .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
+                    .join(' ');
```

Comment:
seems like there's a little bit of inconsistency with the indenting on methods that are added on a newline, can all of these be indented the usual 4 spaces extra so:

```
var title = srcDirPath
    .split(path.sep)
    .pop()
    // etc.
```